pymc/step_methods/compound.py

```diff
@@ -164,3 +163,43 @@ def reset_tuning(self):
     @property
     def vars(self):
         return [var for method in self.methods for var in method.vars]
+
+
+def flatten_steps(step: Union[BlockedStep, CompoundStep]) -> List[BlockedStep]:
+    """Flatten a hierarchy of step methods to a list."""
+    if isinstance(step, BlockedStep):
+        return [step]
+    steps = []
+    if not isinstance(step, CompoundStep):
+        raise ValueError(f"Unexpected type of step method: {step}")
+    for sm in step.methods:
+        steps += flatten_steps(sm)
+    return steps
+
+
+class StatsBijection:
+    """Map between a `list` of stats to `dict` of stats."""
+
+    def __init__(self, sampler_stats_dtypes: Sequence[Dict[str, type]]) -> None:
+        # Keep a list of flat vs. original stat names
+        self._stat_groups: List[List[Tuple[str, str]]] = [
+            [(f"sampler_{s}__{statname}", statname) for statname, _ in names_dtypes.items()]
+            for s, names_dtypes in enumerate(sampler_stats_dtypes)
+        ]
+
+    def map(self, stats_list: StatsType) -> StatsDict:
```

Comment:
Are `map` and `rmap` dedicated or established jargon for this kind of operation? If not, what would you think of using more explicit method names such as `as_dict` (or `to_dict` or something similar) for `map`, and `as_list` for `rmap`?

Reply:
I wanted to align with the `DictToArrayBijection`, but otherwise I would agree